backend/spec/controllers/spree/admin/refunds_controller_spec.rb

```diff
@@ -4,18 +4,36 @@
   stub_authorization!
 
   describe "POST create" do
-    context "a Spree::Core::GatewayError is raised" do
-      let(:payment) { create(:payment) }
+    let(:refund_reason) { create(:refund_reason) }
+    let(:refund_amount) { 100.0 }
+
+    let(:payment) { create(:payment, amount: payment_amount) }
+    let(:payment_amount) { refund_amount * 2 }
 
-      subject do
-        post :create,
-          params: {
-          refund: { amount: "50.0", refund_reason_id: "1" },
+    subject do
+      post :create,
+        params: {
+          refund: {
+            amount: refund_amount,
+            refund_reason_id: refund_reason.id,
+            transaction_id: nil
+          },
           order_id: payment.order_id,
           payment_id: payment.id
         }
-      end
+    end
 
+    it "creates a refund record" do
+      expect{ subject }.to change(Spree::Refund, :count).by(1)
+    end
+
+    it "calls #perform!" do
+      subject
+      # transaction_id comes from Spree::Gateway::Bogus.credit
+      expect(Spree::Refund.last.transaction_id).to eq("12345")
+    end
+
+    context "a Spree::Core::GatewayError is raised" do
```

Comment:
Could we please also check here that no `Refund` object is created when a failure occurs?